rmf_site_editor/src/widgets/inspector/inspect_model_description/inspect_robot_properties.rs

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (C) 2025 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use super::{get_selected_description_entity, ModelDescriptionInspector};
+use crate::{
+    site::{
+        update_model_instances, Affiliation, ChangePlugin, Group, ModelMarker, ModelProperty, Robot,
+    },
+    widgets::{prelude::*, Inspect},
+    ModelPropertyData,
+};
+use bevy::{ecs::system::SystemParam, prelude::*};
+use bevy_egui::egui::{ComboBox, Ui};
+use rmf_site_format::RobotProperty;
+use smallvec::SmallVec;
+use std::collections::HashMap;
+
+pub type ShowRobotPropertyWidgetFn = fn(&mut serde_json::Value, &mut Ui);
+
+/// This resource keeps track of all the properties that can be configured for a robot.
+#[derive(Resource)]
+pub struct RobotPropertyData(pub HashMap<String, HashMap<String, ShowRobotPropertyWidgetFn>>);
```

Comment:
Nitpick: Maybe rename this to `RobotPropertyWidgets`. `Data` doesn't seem like the right descriptor to me.